R/rows.R

```diff
@@ -775,6 +775,27 @@ rows_insert_prep <- function(con, x_name, y, by, lvl = 0) {
   out
 }
 
+rows_auto_copy <- function(x, y, copy) {
+  name <- remote_name(x)
+  x_types <- get_col_types(remote_con(x), name)[colnames(y)]
+  auto_copy(x, y, copy = copy, types = x_types)
+}
+
+get_col_types <- function(con, name) {
+  if (inherits(con, "TestConnection")) {
+    return(NULL)
+  }
+  if (is_null(name)) {
+    return(NULL)
+  }
+  res <- DBI::dbSendQuery(con, paste0("SELECT * FROM ", name))
+  on.exit(DBI::dbClearResult(res))
+  DBI::dbFetch(res, n = 0)
+  out <- DBI::dbColumnInfo(res)
```

Comment:
```suggestion
  res <- DBI::dbGetQuery(con, paste0("SELECT * FROM ", name), n = 0)
  out <- DBI::dbColumnInfo(res)
```
?

Reply:
Unfortunately, this does not work. `dbColumnInfo()` needs a DBIResult, not a dataframe.